frontend/src/components/pages/RoutePlanningPage.tsx

```diff
@@ -26,32 +37,90 @@ const MapWrapper = () => {
 
   return (
     <div style={style}>
-      <MapView />
+      <MapView
+        onRouteDataChange={onRouteDataChange}
+        initialData={initialData}
+      />
     </div>
   );
 };
 
+interface RouteRecord {
+  id: number;
+  route_name: string;
+  route_data?: string | { points: Point[]; segments: RouteSegment[] };
+}
+
+interface GetRoutesResponse {
+  success: boolean;
+  data?: RouteRecord[];
+}
+
 const RoutePlanningPage = () => {
-  /* id and route_name may be undefined, if the user is creating a new route */
   const [searchParams] = useSearchParams();
   const initialRouteId = parseInt(searchParams.get("id") || "-1");
   const [routeId, setRouteId] = useState<number>(initialRouteId);
   const initialRouteName = searchParams.get("route_name") || "New Route";
```

Comment:
it is preferable that route id is the only parameter that we use searchParams to fetch, route name should follow the method of how we fetch routeData.